ngraph/core/reference/src/runtime/reference/experimental_detectron_roi_feature_extractor.cpp

```diff
@@ -0,0 +1,464 @@
+                                           const int64_t pooled_width,
+                                           const int64_t iy_upper,
+                                           const int64_t ix_upper,
+                                           T roi_start_h,
+                                           T roi_start_w,
+                                           T bin_size_h,
+                                           T bin_size_w,
+                                           int64_t roi_bin_grid_h,
+                                           int64_t roi_bin_grid_w,
+                                           std::vector<PreCalc<T>>& pre_calc)
+    {
+        int64_t pre_calc_index = 0;
+        for (int64_t ph = 0; ph < pooled_height; ph++)
+        {
+            for (int64_t pw = 0; pw < pooled_width; pw++)
+            {
+                for (int64_t iy = 0; iy < iy_upper; iy++)
+                {
+                    const T yy = roi_start_h + ph * bin_size_h +
+                                 static_cast<T>(iy + 0.5f) * bin_size_h /
+                                     static_cast<T>(roi_bin_grid_h); // e.g., 0.5, 1.5
+                    for (int64_t ix = 0; ix < ix_upper; ix++)
+                    {
+                        const T xx =
+                            roi_start_w + pw * bin_size_w +
+                            static_cast<T>(ix + 0.5f) * bin_size_w / static_cast<T>(roi_bin_grid_w);
+
+                        T x = xx;
+                        T y = yy;
+                        // deal with: inverse elements are out of feature map boundary
+                        if (y < -1.0 || y > height || x < -1.0 || x > width)
+                        {
+                            // empty
+                            PreCalc<T> pc;
+                            pc.pos1 = 0;
+                            pc.pos2 = 0;
+                            pc.pos3 = 0;
+                            pc.pos4 = 0;
+                            pc.w1 = 0;
+                            pc.w2 = 0;
+                            pc.w3 = 0;
+                            pc.w4 = 0;
+                            pre_calc.at(pre_calc_index) = pc;
+                            pre_calc_index += 1;
+                            continue;
+                        }
+
+                        if (y <= 0)
+                        {
+                            y = 0;
+                        }
+                        if (x <= 0)
+                        {
+                            x = 0;
+                        }
+
+                        int64_t y_low = static_cast<int64_t>(y);
+                        int64_t x_low = static_cast<int64_t>(x);
+                        int64_t y_high = 0;
+                        int64_t x_high = 0;
+
+                        if (y_low >= height - 1)
+                        {
+                            y_high = y_low = height - 1;
+                            y = (T)y_low;
```

Comment:
Nonce I see `static_cast` once C style cast. Can you stick to C++ casts?

Reply:
Fixed.